src/mongocxx/collection.cpp

```diff
@@ -431,12 +431,85 @@ std::int64_t collection::count(bsoncxx::document::view filter, const options::co
 }
 
 bsoncxx::document::value collection::create_index(bsoncxx::document::view keys,
-                                                  bsoncxx::document::view options) {
+                                                  const options::index& options) {
```

Comment:
With other options moving towards a `to_document` method, I think I'm more in favor of the first option. Having an "extensions" member seems a little messy, and like things could go wrong. I like the idea of "use our nice options class or send arbitrary bson and suffer possible consequences".